xbmc/cores/playercorefactory/PlayerCoreFactory.cpp

```diff
@@ -310,34 +295,32 @@ bool CPlayerCoreFactory::LoadConfiguration(const std::string &file, bool clear)
   TiXmlElement *pConfig = playerCoreFactoryXML.RootElement();
   if (pConfig == NULL)
   {
-      CLog::Log(LOGERROR, "Error loading %s, Bad structure", file.c_str());
-      return false;
+    CLog::Log(LOGERROR, "Error loading %s, Bad structure", file.c_str());
+    return false;
   }
 
   if (clear)
   {
-    for(std::vector<CPlayerCoreConfig *>::iterator it = m_vecCoreConfigs.begin(); it != m_vecCoreConfigs.end(); ++it)
-      delete *it;
-    m_vecCoreConfigs.clear();
-    // Builtin players; hard-coded because re-ordering them would break scripts
-    CPlayerCoreConfig* VideoPlayer = new CPlayerCoreConfig("VideoPlayer", EPC_VideoPlayer, NULL);
-    VideoPlayer->m_bPlaysAudio = VideoPlayer->m_bPlaysVideo = true;
-    m_vecCoreConfigs.push_back(VideoPlayer);
-
-     // Don't remove this, its a placeholder for the old MPlayer core, it would break scripts
-    CPlayerCoreConfig* mplayer = new CPlayerCoreConfig("oldmplayercore", EPC_VideoPlayer, NULL);
-    m_vecCoreConfigs.push_back(mplayer);
-
-    CPlayerCoreConfig* paplayer = new CPlayerCoreConfig("PAPlayer", EPC_PAPLAYER, NULL);
-    paplayer->m_bPlaysAudio = true;
-    m_vecCoreConfigs.push_back(paplayer);
+    for (auto config: m_vecPlayerConfigs)
+      delete config;
+    m_vecPlayerConfigs.clear();
 
-    for(std::vector<CPlayerSelectionRule *>::iterator it = m_vecCoreSelectionRules.begin(); it != m_vecCoreSelectionRules.end(); ++it)
-      delete *it;
+    for (auto rule: m_vecCoreSelectionRules)
+      delete rule;
     m_vecCoreSelectionRules.clear();
+
+    // Builtin players
+    CPlayerCoreConfig* VideoPlayer = new CPlayerCoreConfig("VideoPlayer", "video", nullptr);
+    VideoPlayer->m_bPlaysAudio = true;
+    VideoPlayer->m_bPlaysVideo = true;
+    m_vecPlayerConfigs.push_back(VideoPlayer);
+
+    CPlayerCoreConfig* paplayer = new CPlayerCoreConfig("PAPlayer", "pa", nullptr);
```

Comment:
`music` according to https://github.com/xbmc/xbmc/pull/8525/files#diff-a36830a5e3c435f9ffb6c9b1d7174153R92